src/Serializer/PayloadSerializer.php

```diff
@@ -38,7 +38,7 @@ public function __construct(Options $options)
      */
     public function serialize(Event $event): string
     {
-        if (EventType::transaction() === $event->getType()) {
+        if (EventType::transaction() === $event->getType() || $event->hasAttachments()) {
```

Comment:
This same condition is duplicated in the test and, more importantly, in HttpTransport. Perhaps it could be extracted into a separate helper method, something like `requiresEnvelope()`?